src/Microsoft.DotNet.Cli.Utils/CommandResolution/ProjectToolsCommandResolver.cs

```diff
@@ -323,11 +330,61 @@ private IEnumerable<string> GetPossiblePackageRoots(IProject project)
                 .Build(toolLibrary, null, toolLockFile, framework, null);
 
             var tempDepsFile = Path.GetTempFileName();
-            using (var fileStream = File.Open(tempDepsFile, FileMode.Open, FileAccess.Write))
+
+            var args = new List<string>();
+
+            args.Add(toolDepsJsonGeneratorProject);
+            args.Add($"/p:ProjectAssetsFile=\"{toolLockFile.Path}\"");
+            args.Add($"/p:ToolName={toolLibrary.Name}");
+            args.Add($"/p:ProjectDepsFilePath={tempDepsFile}");
+
+            var toolTargetFramework = toolLockFile.Targets.First().TargetFramework.GetShortFolderName();
+            args.Add($"/p:TargetFramework={toolTargetFramework}");
+
+
+            //  Look for the .props file in the Microsoft.NETCore.App package, until NuGet
+            //  generates .props and .targets files for tool restores (https://github.com/NuGet/Home/issues/5037)
+            var platformLibrary = toolLockFile.Targets
+                .FirstOrDefault(t => framework == t.TargetFramework)
+                ?.GetPlatformLibrary();
+
+            if (platformLibrary != null)
             {
-                var dependencyContextWriter = new DependencyContextWriter();
+                string buildRelativePath = platformLibrary.Build.FirstOrDefault()?.Path;
 
-                dependencyContextWriter.Write(dependencyContext, fileStream);
+                var platformLibraryPath = toolLockFile.GetPackageDirectory(platformLibrary);
+
+                if (platformLibraryPath != null && buildRelativePath != null)
+                {
+                    //  Get rid of "_._" filename
+                    buildRelativePath = Path.GetDirectoryName(buildRelativePath);
+
+                    string platformLibraryBuildFolderPath = Path.Combine(platformLibraryPath, buildRelativePath);
+                    var platformLibraryPropsFile = Directory.GetFiles(platformLibraryBuildFolderPath, "*.props").FirstOrDefault();
```

Comment:
The real solution is for NuGet to generate .props and .targets files for tools that it restores: https://github.com/NuGet/Home/issues/5037.  Then we can just import the files NuGet generates.

This logic is just to get something working until we get that feature from NuGet (though I'm not sure when that will be).